velox/functions/sparksql/String.h

```diff
@@ -845,4 +845,132 @@ struct TranslateFunction {
   }
 };
 
+template <typename T>
+struct ConvFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  static constexpr bool is_default_ascii_behavior = true;
+
+  const uint64_t maxUnsignedInt64_ = 0xFFFFFFFFFFFFFFFF;
+
+  uint64_t parseNumber(
+      const arg_type<Varchar>& input,
+      int32_t fromBase,
+      bool isNegativeInput) {
+    int startIndex = isNegativeInput ? 1 : 0;
+    std::vector<int> numberVector;
+    for (int i = startIndex; i < input.size(); i++) {
+      char c = input.data()[i];
+      int number;
+      if (std::isdigit(c)) {
+        number = c - '0';
+        // Only valid characters will be converted till the first
+        // occurence of invalid character.
+        if (number >= fromBase) {
+          break;
+        }
+        numberVector.push_back(number);
+      } else if (std::islower(c)) {
+        number = c - 'a' + 10;
+        if (number >= fromBase) {
+          break;
+        }
+        numberVector.push_back(number);
+      } else if (std::isupper(c)) {
+        number = c - 'A' + 10;
+        if (number >= fromBase) {
+          break;
+        }
+        numberVector.push_back(number);
+      } else if (c == ' ') {
+        continue;
+      } else {
+        break;
+      }
+    }
+    int base = 1;
+    uint64_t unsignedValue = 0;
+    for (int i = numberVector.size() - 1; i >= 0; i--) {
+      uint64_t originalValue = unsignedValue;
+      unsignedValue += numberVector[i] * base;
+      // Overflow case.
+      if (unsignedValue < originalValue) {
+        return maxUnsignedInt64_;
+      }
+      base *= fromBase;
+    }
+    return unsignedValue;
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& input,
+      int32_t fromBase,
+      int32_t toBase) {
+    if (input.empty()) {
+      return false;
```

Comment:
Can we document this behavior? E.g. what happens when input is empty or not a valid number.

Reply:
Yes, just documented.